liquidity/liquidity.go

```diff
@@ -155,19 +196,26 @@ func (m *Manager) SuggestSwaps(ctx context.Context) (
 		return nil, nil
 	}
 
-	channels, err := m.cfg.Lnd.ListChannels(ctx)
+	// Get the current server side restrictions.
+	outRestrictions, err := m.cfg.LoopOutRestrictions(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	// Get the current server side restrictions.
-	outRestrictions, err := m.cfg.LoopOutRestrictions(ctx)
+	// List our current set of swaps so that we can determine which channels
+	// are already being utilized by swaps.
```

Comment:
Perhaps add comment explaining that there may be race conditions when at the same time swaps are initiated manually. I don't think we need to worry about that now, but is this something that needs to be addressed in the future? Or do we bet on full automation anyway?

Reply:
I think we will need to address it in future, we need to get quotes for each swap from the server which isn't a trivial amount of time. One solution could be to limit to only auto or manual (require disabling autoloop to dispatch a manual swap), or to add some locking which will block new swaps while we're figuring out autoloop?

Reply:
I thinking that blocking new swaps while auto-loop is running can be good enough. Maybe we can also spread out the auto-looping a bit, for example finding the most unbalanced channel, initiating the swap and then sleeping for some time before going on to the next.